### DISCUSSION TITLE
BRouter routing with speed limits

### DISCUSSION THREAD

**Discussion by devemux86** [devemux86/cruiser#441] [Q&A] [answered]
**Speed limits** arrive in BRouter (offline) routing and Cruiser navigation.

Cruiser will use a new BRouter routing data reader, so speed limits will be available.

First in Cruiser's internal BRouter implementation.
(and in external BRouter app after its next update)

BRouter data is updated daily, so the most recent speed limits are always provided.

<img src="https://github.com/devemux86/cruiser/assets/3484020/75df8ff5-2b6d-477f-9667-151bbdd508ac" width="50%"/>

**Comment by devemux86** [comment]
BRouter (offline) routing with **speed limits** is available for testing in:
- https://github.com/devemux86/cruiser/discussions/451

**Reply by bios4** [reply]
Would it be possible to show the speed limits also during the navigation simulation?

**Reply by devemux86** [reply]
> Would it be possible to show the speed limits also during the navigation simulation?

Do you mean the speed limit panel at the top right?
We can display it in simulation too.

**Reply by bios4** [reply]
> Do you mean the speed limit panel at the top right?

No, I'm talking about this:

<img alt="" src="https://github.com/devemux86/cruiser/assets/145460147/6bb06757-2c6d-425c-a0cd-8a36edda5158" with="33%">

**Reply by devemux86** [reply]
Speed limit warning appears when GPS speed exceeds the speed limit by the selected speed threshold (in app settings).
Simulation uses a virtual speed.

To see the current speed limit while driving, there is a specific info panel to enable at top right.

**Reply by bios4** [reply]
So does that mean that the sign of the current speed limit of the street I'm using (like the "30" above) is not displayed all of the time? It will only be shown when I am driving above this limit/the set threshold?

**Reply by devemux86** [reply]
- The panel with `!` at top right (image 1) appears all the time displaying the current speed limit.
The panel option is in navigation display settings (image 2).

- The warning at top center appears when go above the speed limit by the selected threshold.
The threshold option is in navigation advanced settings (image 3).

<img src="https://github.com/devemux86/cruiser/assets/3484020/d4b870cd-4c7e-46db-80cc-b40e93fc93e8" width="33%"/><img src="https://github.com/devemux86/cruiser/assets/3484020/b91ce71d-b2f2-477f-84eb-5c8cd89fc934" width="33%"/><img src="https://github.com/devemux86/cruiser/assets/3484020/406f57ef-533b-4df3-80cd-dfa9513c3c53" width="33%"/>

**Reply by bios4** [reply]
> The panel with ! at top right appears all the time

Which is - depending on how much other panels are displayed in which order and what the current light conditions are - very hard to recognize. Therefore I did not activate it before.

> The warning at top center

I would LOVE to have this displayed "all the time" instead of the small panel on the right.
It could then start to flash, if one is above the set limit...

Is this possible???

**Reply by devemux86** [reply]
@bios4 let's discuss the display of speed limits here:
- https://github.com/devemux86/cruiser/discussions/452

**Comment by devemux86** [comment, marked as answer]
BRouter (internal) routing with speed limits is available in [Cruiser 3.0.41](https://play.google.com/store/apps/details?id=gr.talent.cruiser).

---

It will also be possible with external BRouter app after its next update.
